walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,151 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.lang.Object;
+import java.util.TreeMap;
+import java.util.Set;
+import java.util.Map;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    //The TreeMap stores pairs of key and value. It sorts pairs in ascending order based on their key.
+    //Key is start time of event and value is event. 
+    //So Events will be sorted based on their start time when they get added to the TreeMap.   
+    TreeMap<Integer, Event> eventsList = new TreeMap<Integer, Event>();
+    
+    //Go through events attendees and request attendees 
+    //and add only the important events to TreeMap.
+    //Important events are events that contain any of the attendees in the request attendees. 
+
+    Iterator i = events.iterator();
+    Iterator iter = request.getAttendees().iterator();
+
+    while (i.hasNext()) {
+        Event currEvent = (Event)i.next();
+        //if meeting request only has one attendee
+        if (request.getAttendees().size() == 1) { 
+            Iterator iterAttendee = request.getAttendees().iterator();
+            String oneAttendee = (String)iterAttendee.next();
+            if (currEvent.getAttendees().contains(oneAttendee)) {
+                eventsList.put(currEvent.getWhen().start(), currEvent);
+            }
+        }
+        //if meeting request contains more than one attendee
+        else { 
+            while (iter.hasNext()) {
```

Comment:
You are using the same iter that was instantiated before the while loops, meaning that every time you run this, it is iterating from where it left off last time (aka you are not iterating the entire request's list of attendees)

I'd suggest avoiding the use of iters since they are confusing and instead use forEach loops

```
for (Event event : events) {
  ...
  for (String attendee : request.getAttendees()) {
     ...
}
```